aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -3444,8 +3465,9 @@ static void linalg_lstsq_out_info(
         compute_residuals = at::all(rank == n).item().toBool();
       }
     }
-    if (compute_residuals) {
-      // LAPACK stores residuals data for postprocessing in rows n:(m-n)
+  if (compute_residuals) {
+    // LAPACK stores residuals data for postprocessing in rows n:(m-n)
+    if (solution.size(-2) >= n + (m - n)) {
```

Comment:
why is all this necessary?

Reply:
The tests were exposing some problems and they mainly originated from use of `narrow` in specific tensor cases. So these new conditionals were added to handle those situations. 

Errors like so: 

```
  File "/home/ksm/pytorch/test/test_linalg.py", line 316, in test_linalg_lstsq
    res = torch.linalg.lstsq(a, b, rcond=rcond, driver=driver)
RuntimeError: start (1) + length (1) exceeds dimension size (1)
```

Would it be preferable to revert these changes, run the test workflows, see what fails and go from there?

Reply:
But this path was already working before. I don't understand why should we touch it at all?

Reply:
It's not needed. When adding tests and trying to go through exceptions, I found `narrow` was the main cause. So I added guards against all the `narrow`s in the code. 
I added this because if the second last dimension was less than `n + m - n` (which I now realize is just `m`) then this will throw. 

I tried to produce a scenario where this would throw an exception, but it's always caught earlier on. This check is redundant and can be removed.